src/transformers/models/clip/modeling_flax_clip.py

```diff
@@ -0,0 +1,1074 @@
+                Indices of input sequence tokens in the vocabulary. Padding will be ignored by default should you
+                provide it.
+
+                Indices can be obtained using :class:`~transformers.CLIPTokenizer`. See
+                :meth:`transformers.PreTrainedTokenizer.encode` and :meth:`transformers.PreTrainedTokenizer.__call__`
+                for details.
+
+                `What are input IDs? <../glossary.html#input-ids>`__
+
+        Returns:
+            text_features (:obj:`jax_xla.DeviceArray` of shape :obj:`(batch_size, output_dim`): The text embeddings
+            obtained by applying the projection layer to the pooled output of :class:`~transformers.FlaxCLIPTextModel`.
+        """
+        if position_ids is None:
+            position_ids = jnp.broadcast_to(jnp.arange(jnp.atleast_2d(input_ids).shape[-1]), input_ids.shape)
+
+        if attention_mask is None:
+            attention_mask = jnp.ones_like(input_ids)
+
+        def _get_features(module, input_ids, attention_mask, position_ids):
+            text_outputs = module.text_model(
+                input_ids=input_ids,
+                attention_mask=attention_mask,
+                position_ids=position_ids,
+            )
+            pooled_output = text_outputs[1]
+            text_features = module.text_projection(pooled_output)
+            return text_features
+
+        return self.module.apply(
+            {"params": self.params},
+            jnp.array(input_ids, dtype="i4"),
+            jnp.array(attention_mask, dtype="i4"),
+            jnp.array(position_ids, dtype="i4"),
+            method=_get_features,
+        )
+
+    def get_image_features(self, pixel_values):
+        r"""
+        Args:
+            pixel_values (:obj:`numpy.ndarray` of shape :obj:`(batch_size, num_channels, height, width)`):
+                Pixel values. Padding will be ignored by default should you provide it. Pixel values can be obtained
+                using :class:`~transformers.CLIPFeatureExtractor`. See
+                :meth:`transformers.CLIPFeatureExtractor.__call__` for details.
+
+        Returns:
+            image_features (:obj:`jax_xla.DeviceArray` of shape :obj:`(batch_size, output_dim`): The image embeddings
+            obtained by applying the projection layer to the pooled output of
+            :class:`~transformers.FlaxCLIPVisionModel`
+        """
+        pixel_values = jnp.transpose(pixel_values, (0, 2, 3, 1))
+
+        def _get_features(module, pixel_values):
+            vision_outputs = module.vision_model(pixel_values=pixel_values)
+            pooled_output = vision_outputs[1]  # pooled_output
+            image_features = module.visual_projection(pooled_output)
+            return image_features
+
+        return self.module.apply(
+            {"params": self.params},
+            jnp.array(pixel_values, dtype=jnp.float32),
+            method=_get_features,
```

Comment:
same I think we could provide `train`, `return_dict` here as well no?

Reply:
added the `train` and `rngs` args. These functions are meant to get the image/text features so we don't return all outputs (attentions, hidden_states) from these.

Reply:
ok!